src/bootstrap/bin/main.rs

```diff
@@ -12,5 +12,31 @@ use bootstrap::{Build, Config};
 fn main() {
     let args = env::args().skip(1).collect::<Vec<_>>();
     let config = Config::parse(&args);
+    check_version(&config);
     Build::new(config).build();
 }
+
+fn check_version(config: &Config) {
+    const VERSION: usize = 0;
+
+    let needs_warning = if let Some(version) = config.version {
+        if version != VERSION {
+            println!(
+                "warning: there have been changes to x.py since you last updated `version = {}`",
+                version
+            );
+            true
+        } else {
+            false
+        }
+    } else {
+        println!(
+            "warning: x.py has made several changes in the past month you may want to look at"
```

Comment:
Where should the user look at?

Reply:
It prints that a few lines later; see https://github.com/rust-lang/rust/pull/76626#issuecomment-691392248 for an example.

Reply:
Yes, but why does it needs this lines? The few lines below already explained it well.

Reply:
I don't think they do?
```
help: consider looking at the changes in `src/bootstrap/CHANGELOG.md`
note: to silence this warning, add `version = 0` to `config.toml`
```
That doesn't say _what_ changed - this line specifically says that _x.py_ changed.

Reply:
```
warning: there have been changes to x.py since you last updated `version = 0`
help: consider looking at the changes in `src/bootstrap/CHANGELOG.md`
note: to silence this warning, add `version = 1` to `config.toml`
```
Wouldn't user get the same warning if they already have `version = 1` in config.toml? I believe it should show `add version = 0 to` only if version does not exists but `change version = 0 in` if config already exists.

Reply:
> Wouldn't user get the same warning if they already have version = 1 in config.toml?

No, that's what the `version != VERSION` check is doing. I could probably improve the error message though.

Reply:
Ok, now it will say 'update `config.toml` to use `version = 1`' instead.

Reply:
Example output:
```
warning: there have been changes to x.py since you last updated `version = 1`
help: consider looking at the changes in `src/bootstrap/CHANGELOG.md`
note: to silence this warning, update `config.toml` to use `version = 0` instead
```